src/persist-client/src/error.rs

```diff
@@ -182,9 +182,15 @@ impl<T> Determinacy for InvalidUsage<T> {
 #[cfg_attr(any(test, debug_assertions), derive(PartialEq))]
 pub struct CodecMismatch {
     /// The requested (K, V, T, D) codecs.
-    pub(crate) requested: (String, String, String, String),
+    ///
+    /// The last element in the tuple is Some when the name of the codecs match,
+    /// but the concrete types don't: e.g. mz_repr::Timestamp and u64.
+    pub(crate) requested: (String, String, String, String, Option<&'static str>),
```

Comment:
The new degree of freedom doesn't seem to be used here... I assume that's expected and setup for future changes, but FYI in case it is not!

Reply:
yup, known! it got used in the commit that's getting pulled out into a second PR